WordPress/WordPressShare/BlogPickerViewController.swift

```diff
@@ -0,0 +1,135 @@
+        return sites?.count ?? 0
+    }
+    
+    override func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return rowHeight
+    }
+    
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        var cell = tableView.dequeueReusableCellWithIdentifier(reuseIdentifier)
+        if cell == nil {
+            cell = WPTableViewCell(style: .Subtitle, reuseIdentifier: reuseIdentifier)
+            WPStyleGuide.Share.configureBlogTableViewCell(cell!)
+        }
+        
+        let site = sites?[indexPath.row]
+        configureCell(cell!, site: site!)
+        
+        return cell!
+    }
+    
+    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        let site = sites![indexPath.row]
+        onChange?(siteId: site.ID, description: site.name?.characters.count > 0 ? site.name : site.URL.absoluteString.hostname())
+        navigationController?.popViewControllerAnimated(true)
+    }
+    
+    
+    // MARK: - Setup Helpers
+    private func setupView() {
+        title = NSLocalizedString("Site Picker", comment: "Title for the Site Picker")
+        preferredContentSize = UIScreen.mainScreen().bounds.size
+    }
+    
+    private func setupTableView() {
+        // Blur!
+        let blurEffect = UIBlurEffect(style: .Light)
+        tableView.backgroundColor = UIColor.clearColor()
+        tableView.backgroundView = UIVisualEffectView(effect: blurEffect)
+        tableView.separatorEffect = UIVibrancyEffect(forBlurEffect: blurEffect)
+        
+        // Fix: Hide the cellSeparators, when the table is empty
+        tableView.tableFooterView = UIView()
+    }
+    
+    private func setupNoResultsView() {
+        noResultsView = WPNoResultsView()
+        tableView.addSubview(noResultsView)
+    }
+    
+    
+    // MARK: - Private Helpers
+    private func loadSites() {
+        let authDetails = ShareExtensionService.retrieveShareExtensionConfiguration()
+        RequestRouter.bearerToken = authDetails!.oauth2Token as String
+        
+        let service = SiteService()
+        
+        showLoadingView()
+        
+        service.fetchSites { sites, error in
+            dispatch_async(dispatch_get_main_queue()) {
+                self.sites = sites
+                self.tableView.reloadData()
+                self.showEmptySitesIfNeeded()
+            }
+        }
+    }
+    
+    private func configureCell(cell: UITableViewCell, site: Site) {
+        // Site's Details
+        cell.textLabel?.text = site.name
+        cell.detailTextLabel?.text = site.URL.absoluteString.hostname()
+        
+        // Site's Blavatar
+        let placeholderImage = WPStyleGuide.Share.blavatarPlaceholderImage
+        
+        if let siteIconPath = site.icon, siteIconUrl = NSURL(string: siteIconPath) {
```

Comment:
Nitpicky, and I don't think we've ever said one way or another, but should we perform assignments on separate lines?